### PR TITLE
Replace github.com/form3tech-oss/jwt-go with https://github.com/golang-jwt/jwt

### PR DESCRIPTION
**Pull request by Haimantika**
Changed github.com/form3tech-oss/jwt-go to https://github.com/golang-jwt/jwt as per discussions on kubernetes/kubernetes#104007

Signed-off-by: Haimantika Mitra <haimantikamitra@gmail.com>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Review by serathius — APPROVED** [review]

**Comment by serathius** [comment]
Can you squash the commits?

**Comment by Haimantika** [comment]
> Can you squash the commits?

I did it for the first time, can you please review if it is fine?

**Comment by spzala** [comment]
Approved CI. Also, cc @mitake

**Review by spzala — COMMENTED** [review]
thanks @Haimantika  lgtm but please see CI failure on Static Analysis.

**Comment by Haimantika** [comment]
> thanks @Haimantika lgtm but please see CI failure on Static Analysis

Looks like the error was on the go.mod file in the server directory, ran go mod tidy and tried pushing the commit, there seems to be some conflict. Can you please review again?

File: server/go.mod [hunk 1]
```diff
@@ -6,8 +6,12 @@ require (
 	github.com/coreos/go-semver v0.3.0
 	github.com/coreos/go-systemd/v22 v22.3.2
 	github.com/dustin/go-humanize v1.0.0
-	github.com/form3tech-oss/jwt-go v3.2.3+incompatible
+<<<<<<< HEAD
```
Comment (lilic) [inline server/go.mod:9]: There seems to still be an issue with some rebase leftover.

**Review by lilic — COMMENTED** [review]
This change makes sense to me, as it seems more maintained! 

There was one issue in the go.mod and if possible could you pin it against a specific release? Thanks!

Also please squash all commits to one commit.

File: server/go.mod (hunk 1 above)
Comment (Haimantika) [reply server/go.mod:9]: Yeah, accepted the incoming changes. The issue was on this file, let me commit and then squash them.

**Comment by spzala** [comment]
> > thanks @Haimantika lgtm but please see CI failure on Static Analysis
> 
> Looks like the error was on the go.mod file in the server directory, ran go mod tidy and tried pushing the commit, there seems to be some conflict. Can you please review again?

Thanks! Running CI..

**Comment by Haimantika** [comment]
> > > thanks @Haimantika lgtm but please see CI failure on Static Analysis
> > 
> > 
> > Looks like the error was on the go.mod file in the server directory, ran go mod tidy and tried pushing the commit, there seems to be some conflict. Can you please review again?
> 
> Thanks! Running CI..

Was expecting this error, since bill-of-materials.json did not change. I have tried running the updatebom.sh script but that did not change anything. I use Windows, is there any OS specific command that I can use to help fix this?

**Review by mitake — APPROVED** [review]
LGTM, thanks!

**Review by lilic — COMMENTED** [review]
One question, otherwise lgtm

File: server/go.mod [hunk 2]
```diff
@@ -6,8 +6,8 @@ require (
 	github.com/coreos/go-semver v0.3.0
 	github.com/coreos/go-systemd/v22 v22.3.2
 	github.com/dustin/go-humanize v1.0.0
-	github.com/form3tech-oss/jwt-go v3.2.3+incompatible
 	github.com/gogo/protobuf v1.3.2
+	github.com/golang-jwt/jwt v3.2.1+incompatible
```
[the remark this answers is not in the document]
Comment (Haimantika) [reply server/go.mod:10]: Sure! Let me change it and squash.

**Review by lilic — APPROVED** [review]
Thank you!

File: server/go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (Haimantika) [reply server/go.mod]: Seems like did not go good, should I change it back to 3.2.1?

**Comment by Haimantika** [comment]
> > > thanks @Haimantika lgtm but please see CI failure on Static Analysis
> > 
> > 
> > Looks like the error was on the go.mod file in the server directory, ran go mod tidy and tried pushing the commit, there seems to be some conflict. Can you please review again?
> 
> Thanks! Running CI..

Hi, there has been a further update of the version, so I changed the files accordingly, it would be great if you can run the CI.

**Review by spzala — APPROVED** [review]
@Haimantika thanks for quickly addressing all the comments.

**Pull request merged**